### PR TITLE
Allow Selection of Flood and Polarization Efficiencies Workspace by File Path

### LINKED ISSUE #34720: Allow flood and polarization efficiency to be specified via file instead of workspace
**Is your feature request related to a problem? Please describe.**

Request from Becky.

In the ISIS Reflectometry interface it is possible to specify a workspace to use for a flood correction in the experiment settings. This requires the workspace to exist in the ADS before the reduction is run. Saving and re-loading the batch will re-populate the workspace name, but the user must re-load the workspace into the ADS before it can be run. Therefore we are not fully capturing the reduction configuration when we save a batch. 

Becky says that this isn't urgent but it is not ideal that we are not fully saving the state.

**Describe the solution you'd like**

By specifying a file name instead of a workspace name, we can capture the file that was used. As long as this still exists on the system, the reduction can be re-run after re-loading the batch file.

**Additional context**

It would probably be possible to specify just the file name and have Mantid's load algorithm find it in the user directories. I think the implementation would be the same either way so it would be up to the user how to use this. Saving the batch would just save whatever text the user entered.

We need to consider that full path names are not ideal because they are different on different platforms. However, currently saving any history of loaded files will include the full system-specific path, so I am not sure history can be replayed on different systems anyway.

We should consider whether we want to check that the file exists at the GUI level and mark it in error if it is not found; or leave checking until the reduction is run. The former could cause delays on entering the GUI text; the latter could cause errors on every row in the batch. 

**Implementation notes**

We want to avoid re-loading the workspace for each run so we should check it exists before loading. `ReflectometryISISPreprocess` has some functionality to do this for runs so we could do something similar.

### PR DESCRIPTION
**Pull request by cailafinn**
**Description of work.**
Added text entry boxes to allow specification of a filepath for the polarisation efficiencies workspace and the flood corrections workspace. This will speed up the workflow by removing the need to load the file separately, and allows the information to be held in the batch. Making it a more complete saving of the GUI state. 

**To test:**
1. Talk to Conor or Rachel about getting the corrections files. It's not public, so it can't be put here. 
3. Boot up the Refl GUI
4. Switch the instrument to POLREF.
5. Go to the Experiment Settings Tab
6. Enable polarisation corrections 
7. Check the combo box gets enabled on workspace and becomes a text edit on filepath. 
9. Enter the file path to the downloaded workspace
11. Go back to the Runs tab
12. Enter `31835` into the Run column (not the group, the second box down) and `1.0` into the angle column
13. Click process
14. Check that the `IvsLam_31835` workspace group contains a workspace ending with `_--` and `_++`
15. Do similar with the flood workspaces.

Fixes #34720 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Review by rbauststfc — COMMENTED** [review]
The relevant enabling and disabling of the combo boxes is working well, and switching display of the workspace selector and line edit keeps the GUI nice and clean. Loading from a file path or name is generally working well and the corrections are applied as expected. 👍

There were just a few things I noticed while testing.

Flood correction:
- [x] Choosing None from the drop-down and then running a reduction causes an error. My feeling is that it would be good if this was consistent with the way the Polarization Efficiency drop-down works, where selecting None means the correction is not applied.

Polarization efficiencies:
- [x] If I select "Workspace" or "Filepath" from the drop-down but then leave the workspace/filename empty I get error "Correction method is not defined" when running the reduction. When I do this for the flood correction, the correction is simply not applied - it feels like it would be good for the polarization correction to work the same way for consistency.

Both:
- [x] If not too tricky, it might be good to validate the filepath inputs in the GUI, so that the user can see immediately if they have entered an invalid filepath for flood and polarization efficiencies. There is an example of this for the `CalibrationFile` input on the Instrument Settings tab, although I appreciate this is a different input type and so may not be directly relevant here.
- [x] I noticed that when I changed instrument on the Runs tab, clearing of the workspace/filename inputs was a bit inconsistent. For flood correction, any selected workspace is cleared but the filepath input is not. For polarization efficiencies, neither workspace or filepath input are cleared. It would be good to handle this consistently between the two, and I think all should probably revert to default/be cleared on a change in instrument?
- [x] This is a minor thing and may just be my lack of understanding of how things are intended to work, but when I tried to load using a file name it required me to include the file extension for `.nxs` files but not for `.nx5` files. Is this expected?

File: qt/scientific_interfaces/ISISReflectometry/GUI/Experiment/ExperimentPresenter.cpp [hunk 1]
```diff
@@ -154,23 +154,32 @@ BackgroundSubtraction ExperimentPresenter::backgroundSubtractionFromView() {
 }
 
 PolarizationCorrections ExperimentPresenter::polarizationCorrectionsFromView() {
-  auto const correctionsChecked = m_view->getPolarizationCorrectionOption();
+  auto const polCorrOption = m_view->getPolarizationCorrectionOption();
 
-  if (!correctionsChecked) {
+  if (polarizationCorrectionTypeFromString(polCorrOption) == PolarizationCorrectionType::None) {
```
[the remark this answers is not in the document]
Comment (ConorMFinn) [reply qt/scientific_interfaces/ISISReflectometry/GUI/Experiment/ExperimentPresenter.cpp:159]: Thanks for this. There were a lot of little uncertainties and I was starting to get worried about some scope creep, so thought it was worth getting it checked before continuing with the tidying. There are a few things (such as the "Correction method is not defined" issue) that I left out to maintain consistency with the previous behaviour. But would be good to get fixed as part of this. I'll work through these and get it more tidied up.

**Comment by ConorMFinn** [comment]
> This is a minor thing and may just be my lack of understanding of how things are intended to work, but when I tried to load using a file name it required me to include the file extension for .nxs files but not for .nx5 files. Is this expected?

This seems to be something with how RROA takes and handles paths, so I think is out of scope for this PR.

**Review by rbauststfc — COMMENTED** [review]
The recent code changes look good and the functionality feels much more polished and user friendly. It's great that the manual testing documentation is updated in this PR too.

Just one thing I noticed about the file path validation is that the reduction succeeds if you just enter the name of a flood/efficiency file located in a Mantid user directory, however the GUI displays this as invalid. To get the input to display as valid I needed to enter the full file path, even though this wasn't actually needed for the reduction to work. I think it's definitely the right thing that we don't require the user to enter the full file path if it's already in their user directories, so would it be possible for the validation to be consistent with this?

**Review by rbauststfc — APPROVED** [review]
The latest changes to provide accurate validation in the GUI when just a filename is used are working really nicely and the code for it also looks really good. Thanks very much for making the effort to get this last bit working! This all looks ready to go in now 👍

**Pull request merged**